Jenkinsfile

```diff
@@ -11,39 +12,195 @@ def branchOrPR(pr) {
   return pr
 }
 
-def post_comment(text, repository, pr_number) {
-    sh """#!/bin/bash
-        curl -s -H "Authorization: token ${GITHUB_TOKEN}" -X POST -d '{"body": "${text}"}' "https://api.github.com/repos/stan-dev/${repository}/issues/${pr_number}/comments"
-    """
+def checkOs(){
+    if (isUnix()) {
+        def uname = sh script: 'uname', returnStdout: true
+        if (uname.startsWith("Darwin")) {
+            return "macos"
+        }
+        else {
+            return "linux"
+        }
+    }
+    else {
+        return "windows"
+    }
 }
 
-@NonCPS
-def get_results(){
-    def performance_log = currentBuild.rawBuild.getLog(Integer.MAX_VALUE).join('\n')
-    def comment = ""
+def escapeStringForJson(inputString){
+    return inputString.trim().replace("\r","\\r").replace("\n","\\n").replace("\t"," ").replace("\"","\\\"").replace("\\", "\\\\")
+}
 
-    def test_matches = (performance_log =~ /\('(.*)\)/)
-    for(item in test_matches){
-        comment += item[0] + "\\r\\n"
+def mapBuildResult(body){
+
+    def returnMap = [:]
+
+    def benchmarks = (body =~ /\/benchmarks\/(\w+)\/(.*?)', (.*?), (.*?), (.*?), (.*?)\)/)
```

Comment:
Is it possible to just grab the output CSVs rather than using regexes to search the output for specific strings? I suspect that would be easier to maintain. The compare script could even emit its own CSV of comparison results.

Reply:
Well of course I can do that but I've done it this way ( assuming output doesn't change very often ) to manipulate values more easily in an oop way and be able to print them in a very user friendly way ( above table ). Even if the script outputs CSV I would still need to extract it from the build log.
Think about it, if needed I can change no worries.

Reply:
I really like the way it's being printed - I agree that needs some structured data that you have to get from somewhere. My initial guess is that having the compare script output something structured will be easier, because in general I think it's easier build & maintain a map from structured->structured than a structured -> unstructured display text -> structured pipeline.

> Even if the script outputs CSV I would still need to extract it from the build log.

You can't read them in in a Jenkinsfile?